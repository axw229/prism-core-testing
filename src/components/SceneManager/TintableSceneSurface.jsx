// @flow
import React, { PureComponent } from 'react'

import { SCENE_TYPES } from 'constants/globals'
import getBeforeHash from 'src/shared/utils/getBeforeHash.util'

type Props = {
  image: string,
  maskId: string,
  filterId: string,
  width?: string | number,
  height?: string | number,
  type: string,
  children: any,
  // This prop will hardcode the width and height, use this when you need to programtically resize.
  scaleSvg?: booelan
}

class TintableSceneSurface extends PureComponent<Props> {
  static baseClass = 'prism-scene-manager__scene__surface'

  getSvgContents () {
    const { image, maskId, filterId, type } = this.props
    const baseUrl = getBeforeHash(window.location.href)
    const mask = `url("${baseUrl}#${maskId}")`
    const filter = `url("${baseUrl}#${filterId}")`

    switch (type) {
      case SCENE_TYPES.ROOM:
        return <image xlinkHref={image} width='100%' height='100%' mask={mask} filter={filter} />
      case SCENE_TYPES.OBJECT:
      case SCENE_TYPES.AUTOMOTIVE:
        return <rect className='rect' x='0' y='0' width='100%' height='100%' mask={mask} filter={filter} />
    }
  }

  render () {
    const { width, height, scaleSvg } = this.props

    return (
      <svg style={scaleSvg ? { width, height } : null} className={TintableSceneSurface.baseClass}
        viewBox={`0 0 ${width} ${height}`}
        version='1.1'
        xmlns='http://www.w3.org/2000/svg'
        xmlnsXlink='http://www.w3.org/1999/xlink'
        preserveAspectRatio='none'>
        {this.props.children}
        {this.getSvgContents()}
      </svg>
    )
  }
}

export default TintableSceneSurface
