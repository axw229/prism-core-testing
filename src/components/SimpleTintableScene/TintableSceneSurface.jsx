// @flow
import React, { PureComponent } from 'react'

import { SCENE_TYPES } from 'constants/globals'
import getBeforeHash from 'src/shared/utils/getBeforeHash.util'

type Props = {
  image: string,
  maskId: string,
  filterId: string,
  width?: string | number,
  height?: string | number,
  type: string,
  children: any,
  // This prop will hardcode the width and height, use this when you need to programtically resize.
  scaleSvg?: boolean,
  adjustSvgHeight?: boolean
}

class TintableSceneSurface extends PureComponent<Props> {
  static baseClass = 'simple-tintable-scene__scene__surface'

  getSvgContents () {
    const { image, maskId, filterId, type } = this.props
    const baseUrl = getBeforeHash(window.location.href)
    const mask = `url("${baseUrl}#${maskId}")`
    const filter = `url("${baseUrl}#${filterId}")`

    switch (type) {
      case SCENE_TYPES.FAST_MASK:
      case SCENE_TYPES.ROOM:
        return <image xlinkHref={image} width='100%' height='100%' mask={mask} filter={filter} />
      case SCENE_TYPES.OBJECT:
      case SCENE_TYPES.AUTOMOTIVE:
        return <rect className='rect' x='0' y='0' width='100%' height='100%' mask={mask} filter={filter} />
    }
  }

  render () {
    const { width, height, scaleSvg, adjustSvgHeight } = this.props
    const getStyleValues = (shouldScale: boolean, shouldAdjustSvgHeight: boolean, scalingWidth: number, scalingHeight: number) => {
      if (!shouldScale && !shouldAdjustSvgHeight) {
        return null
      }

      const styles = {}

      if (shouldScale) {
        styles.width = scalingWidth
        styles.height = scalingHeight
      }

      if (adjustSvgHeight) {
        styles.height = 'auto'
      }

      return styles
    }

    return (
      <svg style={getStyleValues(scaleSvg, adjustSvgHeight, width, height)} className={TintableSceneSurface.baseClass}
        viewBox={`0 0 ${width} ${height}`}
        version='1.1'
        xmlns='http://www.w3.org/2000/svg'
        xmlnsXlink='http://www.w3.org/1999/xlink'
        preserveAspectRatio='none'>
        {this.props.children}
        {this.getSvgContents()}
      </svg>
    )
  }
}

export default TintableSceneSurface
