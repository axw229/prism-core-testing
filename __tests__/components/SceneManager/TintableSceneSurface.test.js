import React from 'react'
import { shallow } from 'enzyme'
import TintableSceneSurface from 'src/components/SceneManager/TintableSceneSurface'

const defaultProps = {
  width: '100px',
  height: '100px'
}

const createSurface = (props = {}) => {
  return shallow(<TintableSceneSurface {...defaultProps} {...props} />)
}

// rendering test
describe('testing Tintable Surface', () => {
  it('SVG should rendering correctly', () => {
    const wrapper = createSurface()
    expect(wrapper.find('svg').exists()).toBe(true)
  })

  it('image should rendering correctly when type is room', () => {
    const wrapper = createSurface({ type: 'rooms' })
    expect(wrapper.find('image').exists()).toBe(true)
  })

  it('rect should rendering correctly when type is automobile', () => {
    const wrapper = createSurface({ type: 'automotive' })
    expect(wrapper.find('rect').exists()).toBe(true)
  })
})
