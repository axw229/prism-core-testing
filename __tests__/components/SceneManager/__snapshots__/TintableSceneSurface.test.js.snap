// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing component should match snapshot 1`] = `
<svg
  className="prism-scene-manager__scene__surface"
  preserveAspectRatio="none"
  version="1.1"
  viewBox="0 0 100px 100px"
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
/>
`;
